schema/body_schema.go

```diff
@@ -45,6 +45,29 @@ type BodySchema struct {
 	// referenced from still unknown locations during the build of the module
 	// schema.
 	ImpliedOrigins ImpliedOrigins
+
+	// Extensions represents any HCL extensions supported in this body
+	Extensions *BodyExtensions
+}
+
+type BodyExtensions struct {
+	Count         bool // count attribute + count.index refs
+	ForEach       bool // for_each attribute + each.* refs
+	DynamicBlocks bool // dynamic "block-name" w/ content & for_each inside
+	SelfRefs      bool // self.* refs which refer to outermost parent block body
```

Comment:
```suggestion
```
Assuming this PR focuses on just `count`, so we can leave out the other extensions for now?